2022-04-29 03:10:09,356 ERROR: Exception on /Did_you_know [GET] [in C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask\app.py:1457]
Traceback (most recent call last):
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask\app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask\app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask\app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask\app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask_login\utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\app\main\routes.py", line 343, in Did_you_knows
    return render_template('Did_you_know.html', title=_('Did_you_know'), form=form, dyk=dyk.items)
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask\templating.py", line 147, in render_template
    return _render(
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask\templating.py", line 128, in _render
    rv = template.render(context)
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\jinja2\environment.py", line 1291, in render
    self.environment.handle_exception()
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\jinja2\environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\app\templates\Did_you_know.html", line 2, in top-level template code
    {% import 'bootstrap/wtf.html' as wtf %}
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\app\templates\base.html", line 1, in top-level template code
    {% extends 'bootstrap/base.html' %}
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 1, in top-level template code
    {% block doc -%}
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 4, in block 'doc'
    {%- block html %}
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 20, in block 'html'
    {% block body -%}
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 23, in block 'body'
    {% block content -%}
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\app\templates\base.html", line 86, in block 'content'
    {% block current_app_content %}{% endblock %}
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\app\templates\Did_you_know.html", line 31, in block 'current_app_content'
    <a href="{{ url_for('main.edit_Did_you_know', id=journal.id) }}" class="glyphicon glyphicon-pencil"></a>
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\jinja2\environment.py", line 474, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'journal' is undefined
2022-04-29 03:10:52,910 INFO: Microblog startup [in C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\app\__init__.py:74]
2022-04-29 03:10:53,930 ERROR: Exception on /Did_you_know [GET] [in C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask\app.py:1457]
Traceback (most recent call last):
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask\app.py", line 2073, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask\app.py", line 1518, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask\app.py", line 1516, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask\app.py", line 1502, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask_login\utils.py", line 261, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\app\main\routes.py", line 343, in Did_you_knows
    return render_template('Did_you_know.html', title=_('Did_you_know'), form=form, dyk=dyk.items)
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask\templating.py", line 147, in render_template
    return _render(
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask\templating.py", line 128, in _render
    rv = template.render(context)
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\jinja2\environment.py", line 1291, in render
    self.environment.handle_exception()
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\jinja2\environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\app\templates\Did_you_know.html", line 2, in top-level template code
    {% import 'bootstrap/wtf.html' as wtf %}
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\app\templates\base.html", line 1, in top-level template code
    {% extends 'bootstrap/base.html' %}
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 1, in top-level template code
    {% block doc -%}
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 4, in block 'doc'
    {%- block html %}
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 20, in block 'html'
    {% block body -%}
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\flask_bootstrap\templates\bootstrap\base.html", line 23, in block 'body'
    {% block content -%}
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\app\templates\base.html", line 86, in block 'content'
    {% block current_app_content %}{% endblock %}
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\app\templates\Did_you_know.html", line 31, in block 'current_app_content'
    <a href="{{ url_for('main.edit_Did_you_know', id=Did_you_know.id) }}" class="glyphicon glyphicon-pencil"></a>
  File "C:\Users\User\PycharmProjects\Help project\EA_project_A02\EA_project_A02-Alex_Correct\microblog\venv\lib\site-packages\jinja2\environment.py", line 474, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'Did_you_know' is undefined
